types/rethinkdb/index.d.ts

```diff
@@ -268,15 +268,124 @@ declare module "rethinkdb" {
         insert(obj: any, options?: InsertOptions): Operation<WriteResult>;
 
         get<TObjectType extends object>(key: string): Operation<TObjectType | null> & Writeable;
-        getAll(key: string, index?: Index): Sequence; // without index defaults to primary key
+
+        /**
+         * Get all documents matching one or more keys on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
         getAll(...keys: string[]): Sequence;
+        /**
+         * Get all documents matching a key on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
+        getAll(key: string, index?: Index): Sequence; // without index defaults to primary key
+        /**
+         * Get all documents matching 2 or more keys on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
+        getAll(key: string, key2: string, index?: Index): Sequence; // without index defaults to primary key
+        /**
+         * Get all documents matching 2 or more keys on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
+        getAll(key: string, key2: string, key3: string, index?: Index): Sequence; // without index defaults to primary key
+        /**
+         * Get all documents matching 2 or more keys on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
+        getAll(key: string, key2: string, key3: string, key4: string, index?: Index): Sequence; // without index defaults to primary key
+        /**
+         * Get all documents matching 2 or more keys on a simple index; defaults to primary key if no index provided.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/)
+         */
+        getAll(
+            key: string,
+            key2: string,
+            key3: string,
+            key4: string,
+            key5: string,
+            index?: Index
+        ): Sequence; // without index defaults to primary key
+        /**
+         * Get all documents matching a compound index key.
+         * See [getAll](https://www.rethinkdb.com/api/javascript/get_all/) and
+         * [compound indexes](https://www.rethinkdb.com/docs/secondary-indexes/javascript/#compound-indexes)
+         */
+        getAll(compoundKey: string[], index: Index): Sequence; // without index defaults to primary key
```

Comment:
One of overloads with `compoundKey` have `index` optional. Did you mean to have the one required or have all the overloads optional?